libs/s25main/desktops/dskCampaignSelection.cpp

```diff
@@ -0,0 +1,246 @@
+    AddText(ID_PreviewTitle, DrawPoint(640, previewTitleOffsetY), "", COLOR_YELLOW, FontStyle::CENTER, LargeFont);
+
+    int const multilineOffsetY = previewTitleOffsetY + LargeFont->getHeight();
+    int const mutlilineExtentY = columnExtentY - (multilineOffsetY - getColumnOffsetY());
+    AddMultiline(ID_PreviewDescription, DrawPoint(scecondColumnOffsetX, multilineOffsetY),
+                 Extent(secondColumnExtentX, mutlilineExtentY), TextureColor::Grey, NormalFont);
+
+    AddTextButton(ID_Back, DrawPoint(380, 560), Extent(200, 22), TextureColor::Red1, _("Back"), NormalFont);
+    AddTextButton(ID_Next, DrawPoint(590, 560), Extent(200, 22), TextureColor::Green2, _("Continue"), NormalFont);
+
+    showCampaignInfo(false);
+
+    AddTimer(ID_TimerFillCampaignTable, 1ms);
+}
+
+void dskCampaignSelection::Msg_TableChooseItem(const unsigned ctrl_id, const unsigned)
+{
+    if(ctrl_id == ID_Table)
+        WINDOWMANAGER.Switch(std::make_unique<dskCampaignMissionSelection>(csi_, getSelectedCampaignPath()));
+}
+
+void dskCampaignSelection::Msg_TableSelectItem(const unsigned ctrl_id, const boost::optional<unsigned>& selection)
+{
+    if(ctrl_id != ID_Table)
+        return;
+
+    ctrlButton& btContinue = *GetCtrl<ctrlButton>(ID_Next);
+    btContinue.SetEnabled(false);
+    showCampaignInfo(false);
+    campaignImage_ = nullptr;
+
+    if(selection)
+    {
+        ctrlTable& campaingTable = *GetCtrl<ctrlTable>(ctrl_id);
+        const std::string& campaignFolder = campaingTable.GetItemText(*selection, 5);
+
+        CampaignDescription desc;
+        CampaignDataLoader loader(desc, campaignFolder);
+        if(loader.Load())
+        {
+            ctrlMultiline& multiline = *GetCtrl<ctrlMultiline>(ID_PreviewDescription);
+            ctrlText& campaignTitle = *GetCtrl<ctrlText>(ID_PreviewTitle);
+            multiline.Clear();
+            multiline.AddString(desc.longDescription, COLOR_YELLOW);
+            campaignTitle.SetText(desc.name);
+
+            if(!desc.image.empty() && LOADER.LoadFiles({desc.image}))
+            {
+                campaignImage_ = LOADER.GetImageN(ResourceId::make(RTTRCONFIG.ExpandPath(desc.image)), 0);
+            }
+            btContinue.SetEnabled(true);
+            showCampaignInfo(true);
+        }
+    }
+}
+
+void dskCampaignSelection::showCampaignInfo(bool show)
+{
+    GetCtrl<ctrlMultiline>(ID_PreviewDescription)->SetVisible(show);
+    GetCtrl<ctrlText>(ID_PreviewTitle)->SetVisible(show);
+    showCampaignInfo_ = show;
+}
+
+boost::filesystem::path dskCampaignSelection::getSelectedCampaignPath()
+{
+    auto* table = GetCtrl<ctrlTable>(ID_Table);
+    const auto& selection = table->GetSelection();
+    boost::filesystem::path campaingPath = selection ? table->GetItemText(*selection, 5) : boost::filesystem::path();
+    if(campaingPath.empty())
+        throw std::runtime_error("Campaign path is empty.");
+    return campaingPath;
```

Comment:
The column can never be empty, can it?
Maybe rather:
```suggestion
    if(!selection)
        throw std::runtime_error("Campaign path is empty.");
    boost::filesystem::path campaignPath = table->GetItemText(*selection, 5);
    Rttr_Assert(!campaignPath.empty());
    return campaignPath;
```